### PR TITLE
fix(flame): failure when accessing `visualViewport.visualViewport` from within `iFrame`

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

Fixes issue where the `Flame` chart was attempting to access [`window.visualViewport`](https://developer.mozilla.org/en-US/docs/Web/API/Window/visualViewport) from within an iframe.

## Details

The `Flame` chart was attempting to access the [`window.visualViewport`](https://developer.mozilla.org/en-US/docs/Web/API/Window/visualViewport) of the top most `window` from an iframe. However this triggered a `DOMException` error as the `visualViewport` property is forbidden expect when accessed by the top `window` context, for cross-origins.

```
DOMException: Failed to read a named property 'visualViewport' from 'Window': Blocked a frame with origin "https://my.kibana.com" from accessing a cross-origin frame.
```

We were only updating this value every `100ms` to capture the latest `pinchZoomScale`. Since this is not accessible from an iframe I elected to simply disable this update loop when in the iframe and access the value in place directly from `window.visualViewport?.scale`;

Now renders fine within iframe 👇🏼 

Zight Recording 2024-08-07 at 01 21 10 PM

## Issues

Related to https://github.com/elastic/kibana/issues/188288

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] Unit tests have been added or updated to match the most common scenarios

### CODE REVIEW

File: packages/charts/src/chart_types/flame_chart/flame_chart.tsx [hunk 1]
```diff
@@ -259,7 +254,7 @@ class FlameComponent extends React.Component<FlameProps> {
 
     // browser pinch zoom handling
     this.pinchZoomSetInterval = NaN;
-    this.pinchZoomScale = browserRootWindow().visualViewport?.scale ?? 1;
+    this.pinchZoomScale = 1; // we set this to the accurate value next
```
Comment (dej611) [inline packages/charts/src/chart_types/flame_chart/flame_chart.tsx:257]: If I get this right the problem arise only for cross-origin `iframe`, not all `iframe` scenarios.
What about wrapping the `browserRootWindow().visualViewport` call into a `try/catch` function wrapper and return `1` as fallback?

Comment (nickofthyme) [reply packages/charts/src/chart_types/flame_chart/flame_chart.tsx]: OK I changed it to test the access and apply the interval if not forbidden. See 707a407

File: packages/charts/src/chart_types/flame_chart/flame_chart.tsx [hunk 2]
```diff
@@ -370,10 +382,15 @@ class FlameComponent extends React.Component<FlameProps> {
...
+  /**
+   * Setup interval to update pinch zoom scale factor
+   *
+   * Note: method accesses `window.visualViewport`
+   */
   private setupViewportScaleChangeListener = () => {
     window.clearInterval(this.pinchZoomSetInterval);
     this.pinchZoomSetInterval = window.setInterval(() => {
-      const pinchZoomScale = browserRootWindow().visualViewport?.scale ?? 1; // not cached, to avoid holding a reference to a `window` object
+      const pinchZoomScale = browserRootWindow().visualViewport?.scale ?? 1;
```
Comment (dej611) [inline packages/charts/src/chart_types/flame_chart/flame_chart.tsx:393]: shouldn't this be wrapper as well?

Comment (nickofthyme) [reply packages/charts/src/chart_types/flame_chart/flame_chart.tsx]: Sure, I put the check at the call site but I guess it makes more sense to add the check in the setup method.

See bf0bd1a

**Review by dej611 — APPROVED** [review]

**Pull request merged**